network/src/dms.rs

```diff
@@ -80,35 +223,58 @@ trait DistributedMessageSetRpcInterface: Send + Sync + 'static {
 /// - If the given directory is locked (possibly by another instance of `DistributedMessageSet`),
 /// it will `await` until the lock is released.
 pub struct DistributedMessageSet<N, S> {
-    _storage: Arc<RwLock<S>>,
+    storage: Arc<RwLock<S>>,
+    config: Config,
     _marker: std::marker::PhantomData<N>,
 }
 
-impl<N: P2PNetwork, S: Storage> DistributedMessageSet<N, S> {
+#[derive(Debug, Clone, Serialize, Deserialize)]
+pub struct Config {
+    /// The interval of the broadcasts.
+    /// If none, it will broadcast only in `add_message()`, not in `serve()`.
+    pub broadcast_interval: Option<Duration>,
+    /// The interval of the direct-peer fetch. If none, it will fetch only in `fetch()`, not in `serve()`.
+    pub fetch_interval: Option<Duration>,
+}
+
+impl<N: GossipNetwork, S: Storage> DistributedMessageSet<N, S> {
     /// Creates a new and empty storage with the given directory.
     /// If there is already a directory, it discards everything and creates a new one.
     /// You should try `open()` first!
     ///
     /// - `dms_key`: The unique key for distinguishing the DMS.
```

Comment:
```suggestion
    /// - `dms_key`: The key for distinguishing the type of the DMS instance. (e.g. governance, consensus, ...)
```
Suggest improved comment

Reply:
updated

Reply:
checked